packages/aws-cdk/lib/init-templates/app/dotnet/src/HelloCdk/HelloConstruct.cs

```diff
@@ -0,0 +1,29 @@
+using Amazon.CDK;
+using Amazon.CDK.AWS.IAM;
+using Amazon.CDK.AWS.S3;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace HelloCdk
+{
+    public class HelloConstruct : Construct
+    {
+        readonly IList<Bucket> _buckets = new List<Bucket>();
+
+        public HelloConstruct(Construct parent, string id, HelloConstructProps props) : base(parent, id)
+        {
+            foreach (int i in Enumerable.Range(0, props.BucketCount))
+            {
+                _buckets.Add(new Bucket(this, $"Bucket{i}", new BucketProps()));
+            }
+        }
+
+        public void GrantRead(IIIdentityResource principal)
```

Comment:
`IIIdentityResource`?

Reply:
It was punted to a future release--there are several edge cases to consider (what if `IdentityResource` and `IIdentityResource` are both interfaces? What if an interface has a name like `IAMMyInterface`?). See https://github.com/awslabs/jsii/issues/109.